lib/Event/Model/Asset/ResolveUploadTargetEvent.php

```diff
@@ -25,20 +25,28 @@ class ResolveUploadTargetEvent extends Event
 
     protected string $filename;
 
-    protected array $context;
+    protected array $context = [];
 
     protected int $parentId;
 
     /**
      * ResolveUploadTargetEvent constructor.
      *
-     * @param array $context contextual information
+     * @param array|null $context contextual information
      */
-    public function __construct(int $parentId, string $filename, array $context)
+    public function __construct(int $parentId, string $filename, ?array $context)
     {
         $this->parentId = $parentId;
         $this->filename = $filename;
-        $this->context = $context;
+        if ($context) {
```

Comment:
I think this condition should be more strict, otherwise there will be no deprecation if someone passes an empty context.

```suggestion
        if ($context !== null) {
```